### PR TITLE
Add basic unit test framework

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

**Review by marcofranssen — APPROVED** [review]
LGTM 🚀

File: .github/workflows/helm-chart-ci.yaml [hunk 1]
```diff
@@ -69,6 +69,16 @@ jobs:
             exit 1
           fi
 
+      - name: Setup Go
+        uses: actions/setup-go@v4
```
Comment (marcofranssen) [inline .github/workflows/helm-chart-ci.yaml:73]:
```suggestion
        uses: actions/setup-go@v4.0.1
        with:
          go-version-file: tests/go.mod
          cache-dependency-path: tests/go.sum
          check-latest: true

```

File: .github/workflows/helm-chart-ci.yaml [hunk 2]
```diff
@@ -69,6 +69,16 @@ jobs:
...
+      - name: Run Unit Tests
+        run: |
+          set +e
+          set -o pipefail
+          go install github.com/onsi/ginkgo/v2/ginkgo@latest
+          cd tests/unit
+          ginkgo
```
Comment (marcofranssen) [inline .github/workflows/helm-chart-ci.yaml:80]:
```suggestion
        run: ginkgo
        working-directory: ./tests/unit
```

File: tests/unit/spire_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		return nil, err
+	}
+	objs, err := helmengine.Render(chart, v)
+	return objs, err
+}
+
+var _ = Describe("Spire", func() {
+	chart, err := helmloader.Load("../../charts/spire")
+	if err != nil {
+		panic(err)
```
Comment (faisal-memon) [inline tests/unit/spire_test.go:30]: Is panic on error normal for this unit test framework? Or is there an `Expect(err).ToBeNil` or something like that?

Comment (marcofranssen) [reply tests/unit/spire_test.go:30]: Ah yes for other framework I normally use there is a `Expect.NoError(err)`. Probably nicer and might do some nice logging as well.

Comment (kfox1111) [reply tests/unit/spire_test.go:30]: ./spire_test.go:29:9: Expect.NoError undefined (type func(actual interface{}, extra ...interface{}) "github.com/onsi/gomega/types".Assertion has no field or method NoError)...

Comment (kfox1111) [reply tests/unit/spire_test.go:30]: Docs make it sound like this should do the trick:
        Expect(err).Should(Succeed())

Comment (faisal-memon) [reply tests/unit/spire_test.go:30]: How about `Expect(err).ShouldNot(HaveOccurred())`

File: tests/unit/spire_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+			objs, err := ValueStringRender(chart, `
+spire-server:
+  upstreamAuthority:
+    certManager:
+      enabled: true
+      issuer_name: abc123
+`)
+			if err != nil {
+				panic(err)
+			}
```
Comment (faisal-memon) [inline tests/unit/spire_test.go]:
```suggestion
			Expect(err).Should(Succeed())
```

**Pull request merged**